.github/workflows/build.yml

```diff
@@ -149,6 +154,20 @@ jobs:
         uses: docker/build-push-action@v4
         with:
           context: .
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
           push: ${{ github.event_name == 'release' }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha,scope=mkdocs-docker-image
+          cache-to: type=gha,scope=mkdocs-docker-image,mode=max
```

Comment:
Also adds cache support for Docker build which should reduce build time for incremental builds but we need to keep `__pycache__` and `/root/.cache` and rework a bit the Dockerfile.

Reply:
Added extra commit to use [multi-stage builds](https://docs.docker.com/build/building/multi-stage/) in the Dockerfile so cache is more efficient and we don't need to remove temp folders anymore. It also reduces final image size and layers.